egs/aishell2/s5/run.sh

```diff
@@ -20,26 +20,35 @@ tst_set=
 
 nj=20
 stage=1
+mode=normal        # whether we wanna train simple or normal models. The 'normal' chain
+                   # model includes pitch feats, i-vector and dropout, having stronger
+                   # baseline results in terms of CER. Should be either 'normal' or 'simple'
 gmm_stage=1
 
 . ./cmd.sh
 . ./path.sh
 . ./utils/parse_options.sh
 
+# check mode option
+[ "$mode" != "normal" ] && [ "$mode" != "simple" ] && \
+  echo "mode should be either 'normal' or 'simple'" && exit 0;
+
 # prepare trn/dev/tst data, lexicon, lang etc
 if [ $stage -le 1 ]; then
   local/prepare_all.sh ${trn_set} ${dev_set} ${tst_set} || exit 1;
 fi
 
 # GMM
 if [ $stage -le 2 ]; then
-  local/run_gmm.sh --nj $nj --stage $gmm_stage
+  local/run_gmm.sh --nj $nj --stage $gmm_stage --mode $mode
 fi
 
 # nnet3 & chain
+# so far _1d is the SoTA chain recipe for normal use. If better model get
+# trained & tested it will hopefully be manually updated in later PRs
 if [ $stage -le 3 ]; then
-  # local/nnet3/run_tdnn.sh --nj $nj
-  local/chain/run_tdnn.sh --nj $nj
+  [ $mode == "simple" ] && local/chain/run_tdnn.sh --nj $nj --stage 5 || \
```

Comment:
This should be an if-statement.
But also-- you're really not supposed to go into the tuning directory like that.  If there are going to be two sequences of training scripts, one for simple, one for "normal", there should be two soft links, pointing to two sequences of  numbered scripts.   But this gets very complicated.

I general, I'm not really excited about maintaining two different versions of this recipe, especially if mixed up like this.  I'd much rather have just one version-- it's easier to maintain.  Maybe you can just mention the older script and say that at that time we weren't using pitch and ivectors, and the difference is such-and-such?

Also, since you are changing the recipe in this PR, why aren't the results changing?  Were they previously run with these changes, but the changes just weren't checked in?

Reply:
Thanks for your reply! w.r.t your questions:

1) Thanks for pointing out and I'll make fixes abut if and soft-linking with chatting with @dophist ;

2) We are doing this since it depends on actual needs from the user, in terms of whether they wanna use pitch+ivec+dropout or not. So we wanna keep both versions even though _1d does have better results. @dophist perhaps has more comments to tell;

3) We are not maintaining a 'RESULTS' file so there is no global results about GMM model decoding results. If one wanna run _1a, he/she should run in 'simple' mode and the result of chain is in the script local/chain/tuning/run_tdnn_1a.sh; correspondingly, if one wanna run _1d, he/she should use pitch feats from the beginning of mono training ('normal' mode). So the results in the chain scripts are not changing because they are using diff. type of feats from the beginning (i.e. they previously run with these changes and the changes are actually checking in w.r.t the mode option).

Hope it helps without any potentially offensive statement. If so please point it out. Thanks~